src/pages/api/add-participant.ts

```diff
@@ -0,0 +1,31 @@
+import { NextApiRequest, NextApiResponse } from 'next';
+import { getPersonByDiscordId, createPerson } from '../../lib/api';
+import { postClient } from '../../lib/sanity';
+
+export default async function post(req: NextApiRequest, res: NextApiResponse) {
+  const { body } = req;
+
+  let user = await getPersonByDiscordId(body.discordUser);
+
+  if (!user) {
+    user = await createPerson({
+      username: { current: body.discordUser },
+    });
+  }
+
+  try {
+    postClient
```

Comment:
Can we put this logic block on a custom function that lives on `lib/api`, same way as `createReactGroup`?